### PR TITLE
fix: prevent unauthorised remove-users access

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Prevents unauthorised users from removing users from a given organisation. 

## How did you test this code?

Added unit tests.

### CODE REVIEW

File: api/organisations/permissions/permissions.py [hunk 1]
```diff
@@ -88,6 +88,17 @@ class OrganisationPermission(BasePermission):
     def has_permission(self, request, view):
         if view.action == "create" and settings.RESTRICT_ORG_CREATE_TO_SUPERUSERS:
             return request.user.is_superuser
+
+        organisation_id = view.kwargs.get("pk")
+        if organisation_id and not organisation_id.isnumeric():
+            raise APIException("Invalid organisation ID")
```
Comment (khvn26) [inline api/organisations/permissions/permissions.py]: This assumes the 500 status code. Why not use [`ValidationError`](https://www.django-rest-framework.org/api-guide/exceptions/#validationerror) instead?

File: api/tests/unit/organisations/test_unit_organisations_views.py [hunk 2]
```diff
@@ -1797,3 +1797,49 @@ def test_doesnt_retrieve_stale_api_usage_notifications(
     # Then
     assert response.status_code == status.HTTP_200_OK
     assert len(response.data["results"]) == 0
+
```
Comment (khvn26) [inline api/tests/unit/organisations/test_unit_organisations_views.py:1800]: Add a test for non-numeric org id.

**Review by khvn26 — CHANGES_REQUESTED** [review]

File: api/organisations/permissions/permissions.py (hunk 1 above)
Comment (matthewelwell) [reply api/organisations/permissions/permissions.py]: Whoops yep good catch, I wanted 400.

Comment (matthewelwell) [reply api/organisations/permissions/permissions.py]: Done.

File: api/tests/unit/organisations/test_unit_organisations_views.py (hunk 2 above)
Comment (matthewelwell) [reply api/tests/unit/organisations/test_unit_organisations_views.py:1800]: Done.

File: api/organisations/permissions/permissions.py [hunk 3]
```diff
@@ -88,6 +88,17 @@ class OrganisationPermission(BasePermission):
     def has_permission(self, request, view):
         if view.action == "create" and settings.RESTRICT_ORG_CREATE_TO_SUPERUSERS:
             return request.user.is_superuser
+
+        organisation_id = view.kwargs.get("pk")
+        if organisation_id and not organisation_id.isnumeric():
+            raise ValidationError("Invalid organisation ID")
+
+        if view.action == "remove_users":
+            return request.user.is_organisation_admin(int(organisation_id))
```
Comment (khvn26) [inline api/organisations/permissions/permissions.py:97]: Sorry - this will crash the view if `organisation_id` is `None`.

Comment (matthewelwell) [reply api/organisations/permissions/permissions.py:97]: Which it can't be if `view.action == "remove_users"` - it's a detail route.

Comment (khvn26) [reply api/organisations/permissions/permissions.py:97]: What's the point of `if organisation_id` on L93 then?

File: api/organisations/permissions/permissions.py [hunk 4]
```diff
@@ -88,6 +88,17 @@ class OrganisationPermission(BasePermission):
     def has_permission(self, request, view):
         if view.action == "create" and settings.RESTRICT_ORG_CREATE_TO_SUPERUSERS:
             return request.user.is_superuser
+
+        organisation_id = view.kwargs.get("pk")
+        if organisation_id and not organisation_id.isnumeric():
```
Comment (khvn26) [inline api/organisations/permissions/permissions.py:93]: Should this be

```suggestion
        if (not organisation_id) or not organisation_id.isnumeric():
```

?

File: api/organisations/permissions/permissions.py (hunk 3 above)
Comment (khvn26) [reply api/organisations/permissions/permissions.py:97]: See https://github.com/Flagsmith/flagsmith/pull/3791/files#r1569344842.

Comment (matthewelwell) [reply api/organisations/permissions/permissions.py:97]: Because `has_permission` also covers non-retail routes (e.g. create, list, etc.)

**Review by khvn26 — APPROVED** [review]

**Pull request merged**